modules/instrument_manager/php/instrument_manager.class.inc

```diff
@@ -62,58 +62,130 @@ class Instrument_Manager extends \NDB_Menu_Filter
             $sqlSelectOne = "SELECT count(*)".
                             " FROM test_names WHERE Test_name=:testname";
-            $exists       = $db->pselectOne(
+            $count_result = $db->pselectOne(
                 $sqlSelectOne,
-                array('testname' => $_FILES['install_file']['name'])
+                array('testname' => $instname)
             );
-            if ($exists <= 0) {
-                $db_config = $config->getSetting('database');
-                exec(
-                    'php generate_tables_sql_and_testNames.php < '
-                    . escapeshellarg($new_file)
-                );
+            if (intval($count_result) > 0) {
+                $this->tpl_data['feedback'] = "Instrument '$instname' already "
+                    . "exists in the test battery!";
+                http_response_code(409);
+                return;
+            }
 
-                $instrument = \NDB_BVL_Instrument::factory(
-                    $instname,
-                    '',
-                    ''
-                );
-                if (!isset($db_config['quatUser'], $db_config['quatPassword'])) {
-                    throw new \LorisException(
-                        "SQL needs to be sourced manually at /project/tables_sql/".
-                        $instrument->table.".sql"
-                    );
-                }
-                exec(
-                    "mysql".
-                    " -h" . escapeshellarg($db_config['host']).
-                    " -u" . escapeshellarg($db_config['quatUser']).
-                    " -p" . escapeshellarg($db_config['quatPassword']).
-                    " " . escapeshellarg($db_config['database']).
-                    " < " . $this->path . "project/tables_sql/".
-                    escapeshellarg($instrument->table . '.sql')
+            // Don't update the file if it already exists on the back-end.
+            // Instead, inform users that an administrator must install it on
+            // their behalf.
+            // This should only happen for users on a system where automatic
+            // installation is disabled (ie. has no quatUser), as the above
+            // error will return before this one.
+            $file_path = $this->path . "project/instruments/" .
+                        $_FILES['install_file']['name'];
+            if (file_exists($file_path)) {
+                $this->tpl_data['feedback'] = "'$instname.linst' has already "
+                    . "been uploaded. Please contact your administrator to "
+                    . "install the instrument.";
+                http_response_code(409);
+                return;
+            }
+
+            move_uploaded_file($_FILES['install_file']['tmp_name'], $file_path);
+            chmod($file_path, 0644);
+            // Scripts in tools/ often make relative imports, so we must change
+            // our effective directory in order to use them.
+            chdir($this->path . "/tools");
```

Comment:
I don't think this is necessary. htdocs and tools are at the same relative depth.

Reply:
It's the same functionality [as before ](https://github.com/aces/Loris/pull/3746/files#diff-7d481f75a6241f8079ffa0079d2e93baL84); all I did was add a comment and reorder things.

Reply:
Fair enough